cli/tests/unit/serve_test.ts

```diff
@@ -2826,6 +2828,102 @@ Deno.test(
   },
 );
 
+Deno.test(
+  "Throw if disturbed",
+  { permissions: { net: true } },
+  async function shouldThrowIfBodyIsUnusableDisturbed() {
+    const ac = new AbortController();
+    const listeningPromise = deferred();
+
+    function customErrorCb(ac: AbortController): (error: unknown) => Response {
+      return (error) => {
+        assert(error instanceof TypeError);
+        assert(
+          error.message.endsWith(
+            "Body is unusable.",
+          ),
+        );
+                  
+        ac.abort();
+        return new Response("Internal server error", { status: 500 });
+      };
+    }    
+
+    const server = Deno.serve({
+      handler: async (req) => {
+        await req.text();
+
+        req.clone();
+        
+        return new Response("ok");
+      },
+      signal: ac.signal,
+      onListen: ({ port }: { port: number }) => listeningPromise.resolve(port),
+      onError: customErrorCb(ac),
+    });
+    
+    try {
+      const port = await listeningPromise;
+      await fetch(`http://localhost:${port}/`, {
+        headers: { connection: "close" },
+        method: "POST",
+        body: '{"bar":true}'
+      });
+    } catch (e) {
+      assert(e instanceof TypeError);
+    } finally {
+      ac.abort();
+      await server.finished;
+    }
+  },
+);
+
+Deno.test(
+  "Throw if locked",
+  { permissions: { net: true } },
+  async function shouldThrowIfBodyIsUnusableLocked() {
+    const ac = new AbortController();
+    const listeningPromise = deferred();
+
+    function customErrorCb(): (error: unknown) => Response {
+      return (error) => {
+        assert(error instanceof TypeError);
+        assert(
+          error.message.endsWith(
+            "Body is unusable.",
+          ),
+        );
+                  
+        return new Response("Internal server error", { status: 500 });
+      };
+    }    
+
+    const server = Deno.serve({
+      handler: (req) => {
+        const _reader =  req.body?.getReader();
+
+        req.clone();
+        
+        return new Response("ok");
+      },
+      signal: ac.signal,
+      onListen: ({ port }: { port: number }) => listeningPromise.resolve(port),
+      onError: customErrorCb(),
+    });
+    
+    const port = await listeningPromise;
+    const resp = await fetch(`http://localhost:${port}/`, {
+      headers: { connection: "close" },
+      method: "POST",
+      body: '{"bar":true}'
+    });
+
+    ac.abort();
+    await resp.body?.cancel();
+    await server.finished;
+  },
+);
```

Comment:
I'm having some trouble with this test. When I forcefully make it fail in the `customErrorCb` it does not throw and keep passing for some reason. It's very strange to me since the almost identical test above it does throw in the same situation.

Reply:
It's possible that there's a bug here elsewhere in the code. If this test doesn't work as expected with the try/catch as seen above, let's add `ignore: true` and put a comment above it. We can create a follow-up issue to fix whatever that is.